hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/StoreScanner.java

```diff
@@ -351,11 +350,11 @@ public StoreScanner(ScanInfo scanInfo, ScanType scanType,
 
   // Used to instantiate a scanner for compaction in test
   @VisibleForTesting
-  StoreScanner(ScanInfo scanInfo, OptionalInt maxVersions, ScanType scanType,
+  StoreScanner(ScanInfo scanInfo, Integer maxVersions, ScanType scanType,
```

Comment:
Oh, what is the differece between an OptionalInt and an Integer? They are both wrapping types...

Reply:
True both are wrapping but OptionalInt is again wrapping Integer also.
Here we call this:
https://github.com/apache/hbase/pull/673/files#diff-ff286d61ea5468d0f8978181e4f3a896R946
https://github.com/apache/hbase/pull/673/files#diff-ff286d61ea5468d0f8978181e4f3a896R974

At least we can directly pass int value 2 or null rather than OptionalInt.of(2) and Optional.empty()

Reply:
Passing value 2 will introduce a Integer.valueOf(2) as you are passing an Integer, not a int. And Optional.empty is just a static object, which does not make any difference with null...

Reply:
sure, let me change this to as it was before

Reply:
And is it possible to just pass an int? I think not all the int values are valid for maxVersions, maybe we can use -1 to indicate null here?